docs/Cluster-Internal-networks.md

```diff
@@ -0,0 +1,42 @@
+
+## Background
+
+With each OpenShift cluster, internal networks are used for in-cluster communications.  These networks are private networks that rely on ingress and egress Network Address Translation.  If a network does not require external direct routing/exposure, then we can save on IP network reservations by re-using these private networks.  The following networks are the potential private networks for OCP 4:
+
+- Host/Node Network - primary physical interface for each physical host in a cluster (outgoing NAT IP address for all pods on a node)
+- Storage Network - connecting NAS storage (NetApp) to each of the hosts within a cluster.
+- Pod Network - IP addresses assigned to each pod.
+- Service Network - IP addresses assigned to services.
+
+## Network usage
+
+### Host/Node Network
+
+Host IP addresses are used to NAT each hosted PODs outgoing traffic, as well as host specific node/cluster related services.  Incoming application traffic will not be directed at a host IP address.  While we are planning to use outgoing NAT on the F5, there are situations where we may want to be able to direct route to these host IP addresses from inside the data-centres.  Current OCP3 leverages public IP addresses for all host/nodes so a pre-build list of services that would require direct access has not been described/tested yet.
+
+A separate (unique) Host/Node network for each cluster is recommended.
+
+### Storage Network
+
+This network will be leveraged for high speed communication between the NetApp appliance and the cluster Hosts/Nodes.  While no other access is planned for this network, there are still design elements to identify (eg: backup network integration, etc) that may benefit from a unique network assignment.
+
+A separate (unique) Storage network for each cluster is recommended.
+
+### Cluster Pod and Service Networks
+
+These networks are used by internal pod-pod communication while external cluster access (both incoming and outgoing) is routed through the host/node IP address or a routed ingress IP.  This makes these networks good candidates for re-using network ranges.
+
+A known technology that will be impacted by this decision is the Istio Service Mesh implementation.  The internal pod and service networks are used for service discovery within a Mesh backplane.  One design for a multi-cluster deployment uses a single (shared) Istio backplane.  This design requires uniquely identifiable pod endpoints.  A second design uses separate Istio backplanes for each cluster, and replication between backplanes provides cross-cluster service discovery.  This design would NOT require uniquely identifiable pod endpoints between clusters.
+
+Choosing to re-use the internal cluster networks (for pods and services) will remove the option to stretch an Istio Backplane across multiple clusters.  Requesting additional private networks for each production cluster will require requests for reserved network ranges from hosting services.
```

Comment:
How does Istio communicate between clusters? Does each cluster just expose an API endpoint via a route and talk over that?

Reply:
One Istio installation option is to have separate backplane installation (as described here) where the service discover/routing will be done through an Istio Gateway.  The other is where the Istio backplane is stretched across the clusters and service discovery/routing is done directly against the cluster services/pods.  At this point the direction for multi-cluster Istio with Redhat is still in development, and the recommendation reflects a balance between keeping our options open, and reducing our unique network reservation requirements.

Reply:
> service discovery/routing is done directly against the cluster services/pods

What network is this done over? How does one cluster communicate to the other? Is this why the Host Network needs to be routable?